src/actions/guides/database/querying.cr

```diff
@@ -7,294 +7,428 @@ class Guides::Database::Querying < GuideAction
+    `SELECT MAX(users.age) FROM users`
 
     ```crystal
-    post = PostQuery.first
-    # Returns the associated author and does not trigger a preload error
-    post.author!
+    UserQuery.new.select_max(:age)
     ```
 
-    ## Joining and querying associations
+    ## Associations and Joins
 
-    Let's say you have a `Post` model with `has many comments: Comment`, the
-    `Comment` model has a `rating` column that is an integer, and the highest rating
-    is 5 and lowest is 1.
+    ### Joins
 
-    If we want to get all posts that have at least 1 highly rated comment, we
-    could do something like this:
+    ## Preloading
+
+    ### without preloading
+
+    ## No results
+
+    Avram gives you a `none` method to return no results. This can be helpful when under
+    certain conditions you want the resultd to be empty.
 
     ```crystal
-    class PostQuery < Post::BaseQuery
-      def with_highly_rated_comments
-        # Join the comments
-        join_comments
+    UserQuery.new.none
+    ```
 
-        # Get posts with at least one comment with rating greater than or equal to 4
-        comments { |comment_query| comment_query.rating.gte(4) }
+    > This method does not return an empty array immediately. You can still chain other query methods,
+    > but it will always return no records. For example: `UserQuery.new.none.first` will never return a result
 
-        # Or do the same thing with the block shorthand syntax
-        # See "Short one-argument syntax" in https://crystal-lang.org/docs/syntax_and_semantics/blocks_and_procs.html
-        comments(&.rating.gte(4))
+    ## Named Scopes
+
+    Generally it’s best to name your query methods so they are easy to understand and reuse across your app.
+    In Avram, this is done by extracting methods on the query object.
+
+    ```crystal
+    class UserQuery < User::BaseQuery
+      def adults
+        age.gte(18)
+      end
+
+      def search(name)
+        ilike("\#{name}%")
       end
     end
 
-    # Or do the same without making a scope method:
-    PostQuery.new.join_comments.comments(&.rating.gte(4))
+    UserQuery.new.adults.search("Sal")
+    ```
+
+    ## Deleting Records
+
+    ### Delete one
+
+    Deteling a single record is actually done on the [model]() directly. Since each query returns an
+    instance of the model, you can just call `delete` on that record.
+
+    ```crystal
+    user = User.find(4)
+
+    # DELETE FROM users WHERE users.id = 4
+    user.delete
     ```
 
-    These are all the supported join types for associations:
+    ### Delete all
 
-    * `join_{{association_name}}`
-    * `inner_join_{{association_name}}`
-    * `left_join{{association_name}}`
+    If you need to just delete every record in the entire table, you can use `destroy_all` to truncate.
 
-    So if `Project` `has_many tasks : Task` you'd have `join_tasks`,
-    `inner_join_tasks`, etc.
+    `TRUNCATE TABLE users`
 
+    ```crystal
+    UserQuery.new.destroy_all
+    ```
```

Comment:
Might be worth showing an example showing that you can also scope this. UserQuery.new.archived_at.lte(30.days.ago).delete_all

Reply:
I don't think that would work. https://github.com/luckyframework/avram/blob/master/src/avram/query.cr#L6 This method just truncates the whole DB. Or is there another definition somewhere?

Reply:
Hmmm, that is not expected. I will open an issue about this. I would expect there to be a way to delete with a where

Reply:
That method should probably be renamed `truncate` to be super clear https://github.com/luckyframework/avram/issues/110

Reply:
And https://github.com/luckyframework/avram/issues/111

Reply:
But this looks good for now. Might be worth mentioning it isn't chain able and will truncate the table so people aren't surprised